docs/book/v2/dispatching-middleware.md

```diff
@@ -0,0 +1,272 @@
+                    'route'    => '/closure',
+                    'defaults' => [
+                        'controller' => PipeSpec::class,
+                        'middleware' => static function (
+                            ServerRequestInterface $request,
+                            RequestHandlerInterface $handler
+                        ): ResponseInterface {
+                            return $handler->handle($request);
+                        },
+                    ],
+                ],
+            ],
+        ],
+    ],
+];
+```
+
+## Mixed `PipeSpec` Arguments
+
+In this example multiple different possible parameters are passed to `PipeSpec`:
+
+```php
+use Application\Handler\AlbumDetailHandler;
+use Laminas\Mvc\Middleware\PipeSpec;
+use Laminas\Router\Http\Literal;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+return [
+    'router' => [
+        'routes' => [
+            'instance-pipespec-dev-env' => [
+                'type' => Literal::class,
+                'options' => [
+                    'route' => '/instance-pipespec-dev-env',
+                    'defaults' => [
+                        'controller' => PipeSpec::class,
+                        'middleware' => new PipeSpec(
+
+                            // id/alias of service in container
+                            'container_id_for_album_middleware',
+
+                            // anonymous class implementing a MiddlewareInterface
+                            new class implements MiddlewareInterface {
+                                public function process(
+                                    ServerRequestInterface $request,
+                                    RequestHandlerInterface $handler
+                                ): ResponseInterface {
+                                    return $handler->handle($request);
+                                }
+                            },
+
+                            // Closure
+                            static function (
+                                ServerRequestInterface $request,
+                                RequestHandlerInterface $handler
+                            ): ResponseInterface {
+                                return $handler->handle($request);
+                            },
+
+                            // class-string of an Handler which will be resolved by the container
+                            AlbumDetailHandler::class,
+                        ),
+                    ],
+                ],
+            ],
+        ],
+    ],
+];
+```
+
+> ### Note
+>
+> Please note that if you pass instances of middleware or handlers directly, it is not compatible with config caching. You
+> should use that for development only. The recommended variant is passing container managed middleware
+> like `SomeRequestHandler::class`.
```

Comment:
Feature to serialized objects and closures was added to config aggregator and service manager, so this should be slightly adjusted
```suggestion
> Please note that if you pass instances of middleware or handlers directly, it might not compatible with config caching. You
> should use that for development only. The recommended variant is passing container managed middleware
> like `SomeRequestHandler::class`.
```